exposed-jdbc/src/main/kotlin/org/jetbrains/exposed/sql/statements/jdbc/JdbcDatabaseMetadataImpl.kt

```diff
@@ -221,8 +221,9 @@ class JdbcDatabaseMetadataImpl(database: String, val metadata: DatabaseMetaData)
                     rs.close()
                     names
                 }
-                val storedIndexTable = if (tableSchema == currentSchema!!) table.nameInDatabaseCase() else table.nameInDatabaseCaseUnquoted()
-                val rs = metadata.getIndexInfo(catalog, tableSchema, storedIndexTable, false, false)
+
+                val tableNameWithoutSchemaPrefix = table.nameInDatabaseCaseUnquoted()
+                val rs = metadata.getIndexInfo(catalog, tableSchema, tableNameWithoutSchemaPrefix, false, false)
```

Comment:
Regarding some of the failing tests:

Oracle metadata can be tricky. It's odd that, for the same test, `metadata.getPrimaryKeys(catalog, tableSchema, table.nameInDatabaseCaseUnquoted())` accepts and finds the unquoted identifier, but `metadata.getIndexInfo(catalog, tableSchema, table.nameInDatabaseCaseUnquoted(), false, false)` returns `ORA-00942: table or view does not exist`.
I will create a new issue for that so we can take a deeper look into why & what can be done.

For now, please exclude Oracle from these changes by retaining the original logic with an extra condition:
```kt
(tableSchema == currentSchema!! && currentDialect is OracleDialect)
```

Reply:
thanks for the feedback, will do!